src/watcher.js

```diff
@@ -0,0 +1,18 @@
+import chokidar from 'chokidar'
+import { defineModule } from './lib/module.js'
+
+defineModule({
+  name: 'nomen:watcher',
+  dependsOn: ['nomen:builder'],
+  async onLoad(ctx) {
+    if (!ctx.env.NOMEN_DEV) return
+
+    const watcherInstance = chokidar.watch('.', {
+      ignored: [/.nomen/, /node_modules/],
+      awaitWriteFinish: true,
+    })
+
+    watcherInstance.add('./**/*.js')
```

Comment:
this expects all files to be js?

Reply:
that's initial implementation code. 

The final version would be `./**/*.{js,mjs,ts,tsx,jsx}`